In [4]:
m = True
n = 3.14
q = "−7.2345907654"
print (type(m))
print(type(n))
print(type(q))

<class 'bool'>
<class 'float'>
<class 'str'>


In [6]:
String1 = 'Monty Python'
String2 = "Monty Python"
String3 = 'Monty ' + 'Python'

print(String1)
print(String2)
print(String3)

Monty Python
Monty Python
Monty Python


In [12]:
Evennumbers = [2, 4, 6, 8, 10, 12, 14, 16, 18]
Temp = Evennumbers[1]+Evennumbers[3]


print(Evennumbers[1])
print(Evennumbers[3])
print(Evennumbers[7])

print(Temp)

4
8
16
12


In [13]:
Suitcase = ["sunglasses", "hat", "passport", "laptop", "suit", "shoes"]
print(Suitcase[1:]) 
print(Suitcase[:4])


['hat', 'passport', 'laptop', 'suit', 'shoes']
['sunglasses', 'hat', 'passport', 'laptop']


In [14]:
xlist = [1, 2, 3, 4]
for x in xlist:
     print(x + 1)

2
3
4
5


In [15]:
nlist = [8, 92, 79, 55, 23, 17, 4, 45, 63, 9, 100]
threshold = input('Type Threshold: ')
if (max(nlist) > threshold):
     print('Great')
else:
     print('Not great')

Type Threshold: 55


TypeError: '>' not supported between instances of 'int' and 'str'

In [141]:
example1 = ['abab', 'xyz', 'aa', 'x', 'bcb']
example2 = ['', 'x', 'xy', 'xyx', 'xx']
example3 = ['aaa', 'be', 'abc', 'hello']

Suppose that you have a list of strings named examples. Write a Python loop that goes through the list and prints each string where the string length is three or more and the first and last characters of the strings are the same.

Test your code on the following three versions of the list examples:
examples = ['abab', 'xyz', 'aa', 'x', 'bcb']
examples = ['', 'x', 'xy', 'xyx', 'xx']
examples = ['aaa', 'be', 'abc', 'hello']

You may do this exercise either in the interpreter or in a program. Please submit your code and the output from the three examples.

pattern = re.compile('^[^a-z]+$')
import re
for x in examples:
        if float(len(x)) > 2:
            l = x
            print(l)


In [152]:
pattern = re.compile('[^a-z]+$')
patternx = re.compile('^\w+$')
patternz = re.compile('^.+?$')

result = {}
import re
for x in example3:
    if len(x) > 2:
        m = re.search(pattern, x)
        result[x] = m
    else:
        result[x] = "short string"
print(result)
    


    

{'aaa': None, 'be': 'short string', 'abc': None, 'hello': None}
{'aaa': 'TRUE', 'be': 'FALSE', 'abc': 'FALSE', 'hello': 'FALSE'}


In [156]:
resultx = {}
import re
for x in example3:
    if len(x) > 2:
        for i in x:
            if x[0] == x[-1]: 
                resultx[x] = "TRUE"
            else:
                resultx[x] = "FALSE"  
    else:
        resultx[x] = "SHORT"
print(resultx)

{'aaa': 'TRUE', 'be': 'SHORT', 'abc': 'FALSE', 'hello': 'FALSE'}


{'abab': 'FALSE', 'xyz': 'FALSE', 'aa': 'FALSE', 'x': 'FALSE', 'bcb': 'TRUE'}
{'': 'FALSE', 'x': 'FALSE', 'xy': 'FALSE', 'xyx': 'TRUE', 'xx': 'FALSE'}
{'aaa': 'TRUE', 'be': 'FALSE', 'abc': 'FALSE', 'hello': 'FALSE'}

In [86]:
regex = re.compile('\w')
for i in examples:
    for l in i:
        if 
        print(l)

a
b
a
b
x
y
z
a
a
x
b
c
b


TypeError: 'builtin_function_or_method' object is not subscriptable

In [ ]:
for i in examples
    

In [4]:
mp = 'Today is a Great DAY'

mp = mp.lower()
print(mp)
mp = mp.upper()
print(mp)
mp = mp.strip()
print(mp)
print(mp.startswith('X'))  #where X is the starting letter of your string
print(mp)
print(mp.find('Z'))

print(mp)

today is a great day
TODAY IS A GREAT DAY
TODAY IS A GREAT DAY
False
TODAY IS A GREAT DAY
-1
TODAY IS A GREAT DAY


In [20]:
#Use find and string slicing to extract the portions of the string after the colon character, and then use the float function to convert the extracted string into a floating point number with two decimals.

#Submit the code you used and the results in a single file. Once you have submitted your file, check out at least two of your classmates' files and see if their code and results are the same.

str = 'X-DSPAM-Confidence:0.8475'

ssplit = str.split(":")

num = float(ssplit[1])

#print(num)

print('format is {:.2f}'.format(num))

0.8475
format is 0.85


In [28]:
import sys

lone = [10, 20, 30, 40, 50]
ltwo = ["two", "four", "six", "eight", 'ten']
tone = (10, 20, 30, 40, 50)
ttwo = ("two", "four", "six", "eight",'ten')
ltone = [("cheese", "queso"), ("red", "rojo"), ("school", "escuela")]

print("LENGTH")
print( "lone = ", len(lone))
print( "ltwo = ", len(ltwo))
print( "tone = ", len(tone))
print( "ttwo = ", len(ttwo))
print( "ltone = ", len(ltone))

print() 
print()
print("SIZE")
print( "lone = ", sys.getsizeof(lone))
3print( "ltwo = ", sys.getsizeof(ltwo))
print( "tone = ", sys.getsizeof(tone))
print( "ttwo = ", sys.getsizeof(ttwo))
print( "ltone = ", sys.getsizeof(ltone))

LENGTH
lone =  5
ltwo =  5
tone =  5
ttwo =  5
ltone =  3


SIZE
lone =  104
ltwo =  104
tone =  88
ttwo =  88
ltone =  88


In [35]:

#K McKee Week2 Program Exercise
#Purpose: 
#Create a folder on your computer to store your Python programs. 
#Write a simple program to collect the number of hours worked and rate of pay from the user, 
#calculate the pay, output should be 'Pay:' and the amount calculated. Run your program in Python, 
#generate your result, 
#and place your program code and result in a text document and upload it here.

#collect data
hours = input('Input number of hours worked: ')
rate = input('Input bill rate: ')
#find total
total = float(hours)*float(rate)

#print(total)
#print total 
print('The total for the period is:')
print('$'"{0:.2f}".format(total))

Input number of hours worked: 40
Input bill rate: 30.50
The total for the period is:
$1220.00


In [68]:
# This program reads a file and prints the lines and creates a list of items on the line


# open the file for reading (in the same directory as the program)
path = '/Users/kenmckee/Downloads/state_satscores_2004.txt'
SATfile = open (path, 'r')


# iterate over the lines of the file and count the number of lines

count = 0

SATlist = [ ]

for line in SATfile:

    # increment adds one to the count variable

    count += 1

    # strip the newline at the end of the line (and other white space from ends)

    textline = line.strip()

    # split the line on whitespace

    items = textline.split()

    # add the list of items to the SATlist
    
    SATlist.append(items)
# print the number of teams read

print('Schools:', count)


# print the lines from the list

for line in SATlist:
    print ('School:', line)


SATfile.close()



Schools: 51
School: ['New_York', '497', '510']
School: ['Connecticut', '515', '515']
School: ['Massachusetts', '518', '523']
School: ['New_Jersey', '501', '514']
School: ['New_Hampshire', '522', '521']
School: ['D.C.', '489', '476']
School: ['Maine', '505', '501']
School: ['Pennsylvania', '501', '502']
School: ['Delaware', '500', '499']
School: ['Georgia', '494', '493']
School: ['Rhode_Island', '503', '502']
School: ['Virginia', '515', '509']
School: ['North_Carolina', '499', '507']
School: ['Maryland', '511', '515']
School: ['Florida', '499', '499']
School: ['Vermont', '516', '512']
School: ['Indiana', '501', '506']
School: ['South_Carolina', '491', '495']
School: ['Hawaii', '487', '514']
School: ['Oregon', '527', '528']
School: ['Alaska', '518', '514']
School: ['Texas', '493', '499']
School: ['Washington', '528', '531']
School: ['California', '501', '519']
School: ['Nevada', '507', '514']
School: ['Arizona', '523', '524']
School: ['Montana', '537', '539']
School: ['Ohio', '538', '542


For this question, you are to write a program that reads the data in the file state_satscores_2004.txt. Each line of this file has name of a state, mean Verbal SAT score, and mean Math SAT score.

After reading the data, 
a. Print the state with the highest mean Verbal SAT score 
b. Print each state that has a mean Math SAT score greater than 500

Submit the code and the output from your program.

You may use the code developed for the NBA data as a template, but it is absolutely essential that you use appropriate variable names and that you write original comments for what your program does.

If you prefer, you can write this program with a csv reader, using state_satscores_2004.tsv.




In [85]:
num_rows = len(SATlist)



#find schools with math scores > 500
hschm = {}
for (school, verbal, math) in SATlist:
    if float(math) > 500:
        hschm[school] = math

hschv = {}
for (school, verbal, math) in SATlist:
    if float(verbal) > 500:
        hschv[school] = verbal
print("Schools with math above 500")        
print(sorted(hschm.items(), key=lambda kv: kv[1], reverse = True))
print("")
print("Schools with verbal above 500") 
print(sorted(hschv.items(), key=lambda kv: kv[1], reverse = True))
#find the max and the min scores for each category in the list 
print("")
maxmat = max(SATlist, key=lambda x: x[2])
minmat = min(SATlist, key=lambda x: x[2])

maxver = max(SATlist, key=lambda x: x[1])
minver = min(SATlist, key=lambda x: x[1])

print('max math score', maxmat[0], '@', maxmat[2])
print('min math score', minmat[0], '@', minmat[2])

print("")
print('max ver score', maxver[0], '@', maxver[1])
print('min ver score', minver[0], '@', minver[1])




Schools with math above 500
[('Iowa', '602'), ('North_Dakota', '601'), ('Illinois', '597'), ('South_Dakota', '597'), ('Wisconsin', '596'), ('Minnesota', '593'), ('Kansas', '585'), ('Missouri', '585'), ('Nebraska', '576'), ('Michigan', '573'), ('Oklahoma', '566'), ('Louisiana', '561'), ('Tennessee', '557'), ('Kentucky', '557'), ('Utah', '556'), ('Arkansas', '555'), ('Colorado', '553'), ('Alabama', '553'), ('Mississippi', '547'), ('Wyoming', '546'), ('New_Mexico', '543'), ('Ohio', '542'), ('Montana', '539'), ('Idaho', '539'), ('Washington', '531'), ('Oregon', '528'), ('Arizona', '524'), ('Massachusetts', '523'), ('New_Hampshire', '521'), ('California', '519'), ('Connecticut', '515'), ('Maryland', '515'), ('New_Jersey', '514'), ('Hawaii', '514'), ('Alaska', '514'), ('Nevada', '514'), ('West_Virginia', '514'), ('Vermont', '512'), ('New_York', '510'), ('Virginia', '509'), ('North_Carolina', '507'), ('Indiana', '506'), ('Pennsylvania', '502'), ('Rhode_Island', '502'), ('Maine', '501')]

Scho

Schools with math above 500
[('Iowa', '602'), ('North_Dakota', '601'), ('Illinois', '597'), ('South_Dakota', '597'), ('Wisconsin', '596'), ('Minnesota', '593'), ('Kansas', '585'), ('Missouri', '585'), ('Nebraska', '576'), ('Michigan', '573'), ('Oklahoma', '566'), ('Louisiana', '561'), ('Tennessee', '557'), ('Kentucky', '557'), ('Utah', '556'), ('Arkansas', '555'), ('Colorado', '553'), ('Alabama', '553'), ('Mississippi', '547'), ('Wyoming', '546'), ('New_Mexico', '543'), ('Ohio', '542'), ('Montana', '539'), ('Idaho', '539'), ('Washington', '531'), ('Oregon', '528'), ('Arizona', '524'), ('Massachusetts', '523'), ('New_Hampshire', '521'), ('California', '519'), ('Connecticut', '515'), ('Maryland', '515'), ('New_Jersey', '514'), ('Hawaii', '514'), ('Alaska', '514'), ('Nevada', '514'), ('West_Virginia', '514'), ('Vermont', '512'), ('New_York', '510'), ('Virginia', '509'), ('North_Carolina', '507'), ('Indiana', '506'), ('Pennsylvania', '502'), ('Rhode_Island', '502'), ('Maine', '501')]

Schools with verbal above 500
[('South_Dakota', '594'), ('Iowa', '593'), ('Minnesota', '587'), ('Missouri', '587'), ('Wisconsin', '587'), ('Illinois', '585'), ('Kansas', '584'), ('North_Dakota', '582'), ('Nebraska', '569'), ('Oklahoma', '569'), ('Arkansas', '569'), ('Tennessee', '567'), ('Utah', '565'), ('Louisiana', '564'), ('Michigan', '563'), ('Mississippi', '562'), ('Alabama', '560'), ('Kentucky', '559'), ('Colorado', '554'), ('New_Mexico', '554'), ('Wyoming', '551'), ('Idaho', '540'), ('Ohio', '538'), ('Montana', '537'), ('Washington', '528'), ('Oregon', '527'), ('West_Virginia', '524'), ('Arizona', '523'), ('New_Hampshire', '522'), ('Massachusetts', '518'), ('Alaska', '518'), ('Vermont', '516'), ('Connecticut', '515'), ('Virginia', '515'), ('Maryland', '511'), ('Nevada', '507'), ('Maine', '505'), ('Rhode_Island', '503'), ('New_Jersey', '501'), ('Pennsylvania', '501'), ('Indiana', '501'), ('California', '501')]

max math score Iowa @ 602
min math score D.C. @ 476

max ver score South_Dakota @ 594
min ver score Hawaii @ 487